pkg/geo/geomfn/segmentize.go

```diff
@@ -64,10 +64,19 @@ func segmentizeCoords(a geom.Coord, b geom.Coord, maxSegmentLength float64) ([]f
 	// Only 2D distance is considered for determining number of segments.
 	distanceBetweenPoints := math.Sqrt(math.Pow(a.X()-b.X(), 2) + math.Pow(b.Y()-a.Y(), 2))
 
+	doubleNumberOfSegmentsToCreate := math.Ceil(distanceBetweenPoints / maxSegmentLength)
+	doubleNumPoints := float64(len(a)) * (1 + doubleNumberOfSegmentsToCreate)
+	// With doubles, we can exceed the max int, so check that first.
+	if doubleNumPoints > float64(math.MaxInt64) {
+		return nil, errors.Newf(
+			"attempting to segmentize into too many coordinates",
```

Comment:
can we have more info in the message like how line 82 has?

Reply:
Maybe instead we could just move the check on line 80 up and only check it once? Presumably `float64(math.MaxInt64)` will always be greater than `geo.MaxAllowedSplitPoints`.

Reply:
i avoided this because floats are ugly to format, but ok